### PR TITLE
OpenCL beta_lpdf

### PR DESCRIPTION
**Pull request by t4c1**
## Summary

Implements `beta_lpdf` using OpenCL, enabling it to use GPUs.

I had to change return types of OpenCL overload of `size` to make it work with `max_size` (it has to be consistent with scalar overload). I also added requires to `apply_scalar_unary` implementations of `lgamma` and `digamma`. We will probably sooner or later need some requires on all functions that have OpenCL overloads.

## Tests

New implementation of `beta_lpmf` is tested.

## Side Effects

None

## Release notes

Implemented `beta_lpdf` using OpenCL, enabling it to use GPUs.

## Checklist

- [ ] Math issue #2152

- [ ] Copyright holder: Tadej Ciglarič

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

**Comment by t4c1** [comment]
Beta distribution is full expensive `lgamma` and `digamma` calls, so larger speedups are expected:

beta_lpdf

File: stan/math/opencl/prim/size.hpp [hunk 1]
```diff
@@ -9,13 +9,13 @@ namespace math {
 
 /**
  * Returns the size (number of the elements) of a `matrix_cl`.
- * @param a `matric_cl` to determine size of
+ * @param m `matric_cl` to determine size of
```
Comment (rok-cesnovar) [inline stan/math/opencl/prim/size.hpp:12]:
```suggestion
 * @param m `matrix_cl` to determine size of
```

**Review by rok-cesnovar — APPROVED** [review]
One silly little typo fix that is not related to this PR directly. I am sure you have a lot PRs lined up so in order to save Jenkins from a test run you can also do this in another PR.

File: stan/math/opencl/prim/size.hpp (hunk 1 above)
Comment (t4c1) [reply stan/math/opencl/prim/size.hpp:12]: Yeah given how busy Jenkins is lately I will put this in another PR.

**Pull request merged**